### ISSUE TITLE
[OSQuery] Non-compliant ECS field mappings causing conflicts 

### ISSUE THREAD

**Issue by nicpenning** [elastic/integrations#3724]
I am seeing that OSQuery has a handful of conflicting fields for logs-*.

1 example: file.elf.sections is type object for .ds-logs-osquery_manager.result-default-2022.07.13-000001 and for .ds-logs-cassandra.log-default-2022.07.05-000003 that type is nested. Looks at ECS, or appears this should be nested not an object.  https://www.elastic.co/guide/en/ecs/current/ecs-elf.html#field-elf-sections

It appears that the .alerts-ecs-mappings component template is the source for the Cassandra logs which again is nested. whereas logs-osquery_manager.result@package mapping is an object.

Wherever .alerts-ecs-mappings are, it will conflict with OSQuery data and it's mappings.

All of the ones in the image are also in scope, where OSQuery templates/mappings are breaking data views. Version 8.3.2 on the stack and latest integrations as of this day of writing. 

![Screenshot_20220716-225652_Slack](https://user-images.githubusercontent.com/5582679/179383300-d2d70449-08c7-4772-aaea-f921ce318b2e.jpg)

I am not sure where all of these needed to be updated to the correct ECS mappings so I figured I would start here. 😀

**Comment by nicpenning** [comment]
A list of all conflicts:
![image](https://user-images.githubusercontent.com/5582679/179532178-64b5e5cc-ce35-41a1-a6ce-7996ff1fd3f0.png)

Looks like the ones you need to add:
threat.indicator.file.elf.sections
file.elf.sections

**Comment by nicpenning** [comment]
You are welcome!

_Did you have any data in the indicies for osquery manager before you noticed the issue?_ **I did not notice this issues until I installed the OS Query integration and did a few queries to generate data. I don't know if this issue existed before I did my first OSQuery, my guess is no.**
_What were the other field types?_ **Above are all of the field conflicts with the mapping conflicts and field types.**
_With the proposed changed in https://github.com/elastic/integrations/pull/3725 did you happen to test them?_ **I have not been able to test. did not want to alter the system integrations if I didn't have to and was awaiting these changes. I am fairly confident this will fix the issue.**

**Comment by aleksmaus** [comment]
We used a generator internally to generate all kind of mappings and configs 
https://github.com/aleksmaus/osqgen/blob/main/main.go#L336
It was dropping the fields that are not of the type of:
```
m["type"].(string) == "date" ||
			m["type"].(string) == "ip" ||
			m["type"].(string) == "long" ||
			m["type"].(string) == "float" ||
			m["type"].(string) == "boolean" {
			```
Relying on the default keyword "auto-mapping", in order to reduce the size of the mapping (number of explicitly mapped fields) since we already have a ton for osquery itself. 
This has a potential to cause the issue if the first saved document has the field of the numeric value which will create the mapping as "long" instead of the "keyword" for example.

We can either:
1. map all of ECS fields (which will bloat the mapping)
2. maintain the list of the fields that needs to be explicitly mapped due to potential conflict between the value of the field and the mapped type.

I'll update the generator code.

**Comment by zez3** [comment]
Did this fix catch the 8.4.1 release. I still see the conflicts. Or do I have to reindex some indices?
